test/functional/backwards_compatibility.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Downgrade functional test
+
+Test various downgrade scenarios. Compile the previous node binaries:
+
+contrib/devtools/previous_release.sh -f v0.15.1 v0.14.2
+
+Due to RPC changes introduced in v0.14 the below tests won't work for older
+versions without some patches or workarounds.
+"""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    sync_blocks
+)
+
+class DowngradeTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 3
+        self.extra_args = [
+            [],
+            ["-vbparams=segwit:0:0", "-vbparams=csv:0:0"], # v0.15.1
+            ["-vbparams=segwit:0:0", "-vbparams=csv:0:0"] # v0.14.2
+        ]
+
+    def setup_nodes(self):
+        """Override this method to customize test node setup"""
+        self.add_nodes(self.num_nodes, self.extra_args, None, None, [
```

Comment:
Use named args for those unnamed arguments to provide documentation and robustness against future interface changes.

Reply:
Done.